protocol/common/message.go

```diff
@@ -481,6 +483,29 @@ func (v *MentionsAndLinksVisitor) Visit(node ast.Node, entering bool) ast.WalkSt
 	return ast.GoToNext
 }
 
+func containsMentionOf(literal string, mention string) bool {
+	if strings.Contains(literal, mention) {
+
+		if literal == mention {
+			return true
+		}
+
+		literalLen := len(literal)
+		mentionLen := len(mention)
+		idx := strings.Index(literal, mention)
+
+		if strings.HasPrefix(literal, mention) && string(literal[mentionLen]) == " " ||
+			strings.HasSuffix(literal, mention) && idx > 0 && string(literal[idx-1]) == " " {
+			return true
+		}
+
+		if literalLen >= mentionLen+2 && ((idx > 0 && string(literal[idx-1]) == " ") && (string(literal[idx+mentionLen]) == " ")) {
+			return true
+		}
```

Comment:
Maybe a regex would be better here for the optional spaces

Reply:
we should not use regexes for parsing messages as much as possible, as they tend to lead to performance issue (and possible dos), the markdown library we use tries (I believe we do), to pass only once through the string.